### ISSUE TITLE
Feature Request: Note Search

### ISSUE THREAD

**Issue by Kyoani-fan** [emavgl/oinkoin#17]
I added notes to the entries, but if there are too many entries, it's impossible to click on them one by one to find the one I need, so if you can provide a note search function, it will make it easier to use.

**Comment by emavgl** [comment]
Hi! Can you provide me an use case? Do you maybe use note as tags? Because I was thinking to introduce tags that would help with the search /grouping cross categories.

**Comment by Kyoani-fan** [comment]
For example: in the note I record a certain time to go where the consumption record, then I can use the search function to achieve the search keywords in the note to find this record. For example: I went to **market a certain time, added a record, and then wrote the following in the note of the record: two bottles of ** wine in ** market, later I can search ** market to find this record, as well as record entries related to ** market, or search for ** wine to search for this data. 
For the second question, I think using "categories" as category tags for filtering and grouping already meets the requirements. Because the app has implemented a custom add categories, it can be used as a grouping category tag. 
Thank you for your reply!

**Comment by Rex-0x7CB** [comment]
I agree here. There should be a search button that could search through all the fields like notes, title and category. I'm currently using Monefy and it allows me to search through every entry I have created in the last 6 years. It helps me to see if I bought something on a particular day or check the price of something I bought earlier.

Here's a screenshot of Monefy

![Screenshot_Monefy Pro_Edited2](https://github.com/user-attachments/assets/d6cd5cc4-f7d4-4620-a179-e3eed4726b3f)
